cloudify_nsx/library/nsx_common.py

```diff
@@ -307,8 +308,57 @@ def nsx_read(client_session, path, searched_resource, **kwargs):
         if path not in selected_obj:
             return None
         selected_obj = selected_obj[path]
+    else:
+        return selected_obj
+
+
+def nsx_struct_get_list(nsx_object, path):
+    """create path to searched part and convert part to list"""
+    path_list = path.split("/")
+    selected_obj = nsx_object
+    # select parent
+    for path in path_list[:-1]:
+        if path not in selected_obj:
+            selected_obj[path] = {}
+        selected_obj = selected_obj[path]
+
+    # restruct parent
+    last_part = path_list[-1]
+
+    # create if not exist
+    if last_part not in selected_obj:
+        selected_obj[last_part] = []
 
-    return selected_obj
+    # convert dict to list
+    if isinstance(selected_obj[last_part], dict):
+        selected_obj[last_part] = [selected_obj[last_part]]
+
+    return selected_obj[last_part]
+
+
+def nsx_search(client_session, path, name, searched_resource, **kwargs):
+    """search object by name, same logic as in nsx_read
```

Comment:
This docstring seems a bit misleading now since the logic isn't the same (any more?), but instead it uses nsx_read (which is of course good)